velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -382,6 +384,22 @@ parquet::WriterOptions getParquetOptions(
   return parquetOptions;
 }
 
+void Writer::setMemoryReclaimers(
+    const std::shared_ptr<memory::MemoryPool>& pool) {
+  VELOX_CHECK(
+      !pool->isLeaf(),
+      "The root memory pool for parquet writer can't be leaf: {}",
+      pool->name());
+  VELOX_CHECK_NULL(pool->reclaimer());
+
+  if ((pool->parent() == nullptr) || (pool->parent()->reclaimer() == nullptr)) {
+    return;
+  }
+
+  pool->setReclaimer(exec::MemoryReclaimer::create());
```

Comment:
Can you leave a TODO: to trigger parquet stripe flush (or similar mechanism to shrink a parquet writer's memory footprint) when memory arbitrator reclaims memory?